app/Models/User.php

```diff
@@ -848,8 +849,15 @@ static function (QueryBuilder $query): void {
                         ->orderByDesc('updated_at')
                         ->limit(1);
                 }
-            )
-            ->exists();
+            );
+
+        if (true === config('features.docusign-membership-agreement')) {
+            $query->whereHas('envelope', static function (Builder $query): void {
+                $query->where('complete', true);
+            });
+        }
```

Comment:
Does this handle users who signed before DocuSign? Or is it deliberate that it'll force you to re-sign?

Reply:
It deliberately forces re-sign

Reply:
Sounds good.